File: api/app/Models/Schemas/Result.php
<?php

namespace App\Models\Schemas;

use App\Models\Result as Model;

class Result
{
    /**
     * Event title
     * @var string
     * @OA\Property()
     */
    public string $event;

    /**
     * Event date
     * @var string
     * @OA\Property()
     */
    public string $date;

    /**
     * Event year
     * @var int
     * @OA\Property()
     */
    public int $year;

    /**
     * Event location
     * @var string
     * @OA\Property()
     */
    public string $location;

    /**
     * Event country
     * @var string
     * @OA\Property()
     */
    public string $country;

    /**
     * Competition category
     * @var string
     * @OA\Property()
     */
    public string $category;
    
    /**
     * Competition weapon
     * @var string
     * @OA\Property()
     */
    public string $weapon;

    /**
     * Number of entries
     * @var int
     * @OA\Property()
     */
    public int $entries;

    /**
     * Result position
     * @var int
     * @OA\Property()
     */
    public int $position;

    /**
     * Points for position
     * @var double
     * @OA\Property()
     */
    public float $points;

    /**
     * DE points
     * @var double
     * @OA\Property()
     */
    public float $de;

    /**
     * Podium points
     * @var double
     * @OA\Property()
     */
    public float $podium;

    /**
     * Factor
     * @var double
     * @OA\Property()
     */
    public float $factor;

    /**
     * Total points
     * @var double
     * @OA\Property()
     */
    public float $total;


    /**
     * Status
     * @var string
     * @OA\Property()
     */
    public string $status;

    public function __construct(?Model $result)
    {
        if (!empty($result)) {
            $this->entries = $result->result_entry;
            $this->position = $result->result_place;
            $this->points = $result->result_points;
            $this->de = $result->result_de_points;
            $this->podium = $result->result_podium_points;
            $this->total = $result->result_total_points;
            $this->status = $result->result_in_ranking == 'Y' ? 'Y' : 'N';
        }
    }
}

File: mobile/evf/lib/models/result.dart
class Result {
  final String event;
  final int year;
  final String location;
  final String country;
  final String category;
  final String weapon;
  final int entries;
  final int position;
  final double points;
  final double de;
  final double podium;
  final double total;
  final String status;

  Result.fromJson(Map<String, dynamic> doc)
      : event = doc['event'] as String,
        year = doc['year'] as int,
        location = doc['location'] as String,
        country = doc['country'] as String,
        category = doc['category'] as String,
        weapon = doc['weapon'] as String,
        entries = doc['entries'] as int,
        position = doc['position'] as int,
        points = doc['points'] as double,
        de = doc['de'] as double,
        podium = doc['podium'] as double,
        total = doc['total'] as double,
        status = doc['status'] as String;

  Map<String, dynamic> toJson() => {
        "event": event,
        "year": year,
        "location": location,
        "country": country,
        "category": category,
        "weapon": weapon,
        "entries": entries,
        "position": position,
        "points": points,
        "de": de,
        "podium": podium,
        "total": total,
        "status": status
      };
}
